src/EnergyPlus/EarthTube.cc

```diff
@@ -643,50 +624,43 @@ void CalcEarthTube(EnergyPlusData &state)
-                      state.dataEarthTube->EarthTubeSys(Loop).r1) /
-             2.0 / DataGlobalConstants::Pi / state.dataEarthTube->EarthTubeSys(Loop).PipeThermCond;
-        Rs = std::log((state.dataEarthTube->EarthTubeSys(Loop).r1 + state.dataEarthTube->EarthTubeSys(Loop).r2 +
-                       state.dataEarthTube->EarthTubeSys(Loop).r3) /
-                      (state.dataEarthTube->EarthTubeSys(Loop).r1 + state.dataEarthTube->EarthTubeSys(Loop).r2)) /
-             2.0 / DataGlobalConstants::Pi / state.dataEarthTube->EarthTubeSys(Loop).SoilThermCond;
+        Rc = 1.0 / 2.0 / DataGlobalConstants::Pi / thisEarthTube.r1 / PipeHeatTransCoef;
+        Rp = std::log((thisEarthTube.r1 + thisEarthTube.r2) / thisEarthTube.r1) / 2.0 / DataGlobalConstants::Pi / thisEarthTube.PipeThermCond;
+        Rs = std::log((thisEarthTube.r1 + thisEarthTube.r2 + thisEarthTube.r3) / (thisEarthTube.r1 + thisEarthTube.r2)) / 2.0 /
+             DataGlobalConstants::Pi / thisEarthTube.SoilThermCond;
         Rt = Rc + Rp + Rs;
         OverallHeatTransCoef = 1.0 / Rt;
 
         if (AirMassFlowRate * AirSpecHeat == 0.0) {
-            state.dataEarthTube->EarthTubeSys(Loop).InsideAirTemp = GroundTempz1z2t;
+            thisEarthTube.InsideAirTemp = GroundTempz1z2t;
 
         } else {
 
             // Calculation of Pipe Outlet Air Temperature
             if (state.dataEnvrn->OutDryBulbTemp > GroundTempz1z2t) {
                 Process1 = (std::log(std::abs(state.dataEnvrn->OutDryBulbTemp - GroundTempz1z2t)) * AirMassFlowRate * AirSpecHeat -
-                            OverallHeatTransCoef * state.dataEarthTube->EarthTubeSys(Loop).PipeLength) /
+                            OverallHeatTransCoef * thisEarthTube.PipeLength) /
                            (AirMassFlowRate * AirSpecHeat);
-                state.dataEarthTube->EarthTubeSys(Loop).InsideAirTemp = std::exp(Process1) + GroundTempz1z2t;
+                thisEarthTube.InsideAirTemp = std::exp(Process1) + GroundTempz1z2t;
             } else if (state.dataEnvrn->OutDryBulbTemp == GroundTempz1z2t) {
-                state.dataEarthTube->EarthTubeSys(Loop).InsideAirTemp = GroundTempz1z2t;
+                thisEarthTube.InsideAirTemp = GroundTempz1z2t;
             } else {
                 Process1 = (std::log(std::abs(state.dataEnvrn->OutDryBulbTemp - GroundTempz1z2t)) * AirMassFlowRate * AirSpecHeat -
-                            OverallHeatTransCoef * state.dataEarthTube->EarthTubeSys(Loop).PipeLength) /
+                            OverallHeatTransCoef * thisEarthTube.PipeLength) /
                            (AirMassFlowRate * AirSpecHeat);
-                state.dataEarthTube->EarthTubeSys(Loop).InsideAirTemp = GroundTempz1z2t - std::exp(Process1);
+                thisEarthTube.InsideAirTemp = GroundTempz1z2t - std::exp(Process1);
             }
         }
 
-        CalcEarthTubeHumRat(state, Loop, NZ);
+        thisEarthTube.CalcEarthTubeHumRat(state, NZ);
```

Comment:
One nice thing about using references and acting on instances of classes, it starts revealing where we can modify functions to simply be class methods.  :+1:

Reply:
Methods! 🤯